Realm/RLMRealmConfiguration.mm

```diff
@@ -66,6 +66,7 @@
 
 @implementation RLMRealmConfiguration {
     realm::Realm::Config _config;
+    RLMSyncConfiguration *_syncConfiguration;
```

Comment:
If `manualClientResetHandler` is the only thing being used from the sync config we could just store that instead of the entire thing.